src/main/kotlin/org/rust/lang/core/types/ty/TyReference.kt

```diff
@@ -7,12 +7,26 @@ package org.rust.lang.core.types.ty
 
 import org.rust.lang.core.types.infer.TypeFolder
 import org.rust.lang.core.types.infer.TypeVisitor
+import org.rust.lang.core.types.re.Re
 
-data class TyReference(val referenced: Ty, val mutability: Mutability) : Ty(referenced.flags) {
+data class TyReference(
+    val referenced: Ty,
+    val mutability: Mutability,
+    val lifetime: Re
```

Comment:
Maybe it's better to use default parameter value?
`val lifetime: Re = ReUnknown`
And so remove a lot of changes from this PR related to this signature change

Reply:
And let's rename it to `region`